### PR TITLE
AOC-025: Day 8: Memory Maneuver

### PR DESCRIPTION
**Pull request by basseur**
Finished Day 8!

### CODE REVIEW

**Comment by basseur** [comment]
[AOC-025: Day 8: Memory Maneuver](https://trello.com/c/Ev2dpQo2/19-aoc-025-day-8-memory-maneuver)

File: src/main/java/org/basseur/adventofcode/advent2018/days/day08/Node.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    private int[] metadata;
+    /** The child nodes of this node */
+    private Node[] childNodes;
+
+    /**
+     * The constructor parses the nodes array and instantiates its own child nodes, while also storing the meta data
+     *
+     * @param nodes the array of nodes to be parsed
+     */
+    Node(Integer[] nodes) {
```
Comment (mfbieber) [inline src/main/java/org/basseur/adventofcode/advent2018/days/day08/Node.java]: Why Integer and in almost all other cases int? Is there a specific reason here?

Comment (basseur) [reply src/main/java/org/basseur/adventofcode/advent2018/days/day08/Node.java]: `fileReaders.readFileIntoArrayOfIntegers()` returns an `Integer[]` which is then passed as an argument to the constructor of `Node`. If I change it, I need to copy the `Integer[]` to `int[]`. I thought that is just one additional and unnecessary step.

Comment (mfbieber) [reply src/main/java/org/basseur/adventofcode/advent2018/days/day08/Node.java]: Sure, alright, just wanted to understand :)

**Pull request merged**